apps/src/blockly/addons/contextMenu.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Option to undo previous action.
+ * @alias Blockly.ContextMenuItems.registerUndo
+ */
+ const registerDeletable = function() {
+    /** @type {!ContextMenuRegistry.RegistryItem} */
+    const deletableOption = {
+      displayText: function(scope) {
+        // isDeletale is a built in Blockly function that checks whether the block
+        // is deletable, is not a shadow, and if the workspace is readonly.
+        const displayText = scope.block.isDeletable()
+        ? 'Make Undeletable to Users'
+        : 'Make Deletable to Users',
+        return displayText;
+      },
+      preconditionFn: function() {
+        if (Blockly.isStartMode) {
```

Comment:
quick styling thing, you could just do `if (Blockly.isStartmode) return 'enabled';`; curly brackets are not necessary with one line returns, or you could convert this to a ternary operator too: `return Blockly.isStartMode ? 'enabled' : 'hidden';`.